caproto/sync/client.py

```diff
@@ -138,6 +139,7 @@ def make_channel(pv_name, udp_sock, priority, timeout):
             our_role=ca.CLIENT,
             address=address,
             priority=priority)
+        circuit.our_address = udp_sock.getsockname()[:2]
```

Comment:
The address for a Broadcaster's comes from its UDP socket; the address for a VirtualCircuit comes from its TCP socket. Instead of using `udp_sock` you should be using `sockets[chan.circuit]`, which must access after it is defined in L147.